pkg/tsdb/cloudwatch/models/request_parser_test.go

```diff
@@ -124,255 +138,486 @@ func TestRequestParser(t *testing.T) {
-			query.Period = "auto"
 			to := time.Now()
 			from := to.AddDate(-2, 0, 0)
 
-			res, err := parseRequestQuery(query, "ref1", from, to)
+			res, err := ParseMetricDataQueries(query, from, to, false)
 			require.NoError(t, err)
-			assert.Equal(t, 86400, res.Period)
+			require.Len(t, res, 1)
+			assert.Equal(t, 86400, res[0].Period)
 		})
 
 		t.Run("Time range is 2 days, but 16 days ago", func(t *testing.T) {
-			query.Period = "auto"
 			to := time.Now().AddDate(0, 0, -14)
 			from := to.AddDate(0, 0, -2)
-			res, err := parseRequestQuery(query, "ref1", from, to)
+			res, err := ParseMetricDataQueries(query, from, to, false)
 			require.NoError(t, err)
-			assert.Equal(t, 300, res.Period)
+			require.Len(t, res, 1)
+			assert.Equal(t, 300, res[0].Period)
 		})
 
 		t.Run("Time range is 2 days, but 90 days ago", func(t *testing.T) {
-			query.Period = "auto"
 			to := time.Now().AddDate(0, 0, -88)
 			from := to.AddDate(0, 0, -2)
-			res, err := parseRequestQuery(query, "ref1", from, to)
+			res, err := ParseMetricDataQueries(query, from, to, false)
 			require.NoError(t, err)
-			assert.Equal(t, 3600, res.Period)
+			require.Len(t, res, 1)
+			assert.Equal(t, 3600, res[0].Period)
 		})
 
 		t.Run("Time range is 2 days, but 456 days ago", func(t *testing.T) {
-			query.Period = "auto"
 			to := time.Now().AddDate(0, 0, -454)
 			from := to.AddDate(0, 0, -2)
-			res, err := parseRequestQuery(query, "ref1", from, to)
+			res, err := ParseMetricDataQueries(query, from, to, false)
 			require.NoError(t, err)
-			assert.Equal(t, 21600, res.Period)
+			require.Len(t, res, 1)
+			assert.Equal(t, 21600, res[0].Period)
 		})
 	})
+	t.Run("returns error if period is invalid duration", func(t *testing.T) {
+		query := []backend.DataQuery{
+			{
+				JSON: json.RawMessage(`{
+				   "statistic":"Average",
+				   "period":"invalid"
+				}`),
+			},
+		}
+		_, err := ParseMetricDataQueries(query, time.Now().Add(-2*time.Hour), time.Now().Add(-time.Hour), false)
+		require.Error(t, err)
+		assert.Equal(t, `error parsing query "", failed to parse period as duration: time: invalid duration "invalid"`, err.Error())
+	})
 
-	t.Run("Metric query type, metric editor mode and query api mode", func(t *testing.T) {
-		t.Run("when metric query type and metric editor mode is not specified", func(t *testing.T) {
-			t.Run("it should be metric search builder", func(t *testing.T) {
-				query := getBaseJsonQuery()
-				res, err := parseRequestQuery(query, "ref1", time.Now().Add(-2*time.Hour), time.Now().Add(-time.Hour))
-				require.NoError(t, err)
-				assert.Equal(t, MetricQueryTypeSearch, res.MetricQueryType)
-				assert.Equal(t, MetricEditorModeBuilder, res.MetricEditorMode)
-				assert.Equal(t, GMDApiModeMetricStat, res.GetGMDAPIMode())
-			})
+	t.Run("returns parsed duration in seconds", func(t *testing.T) {
+		query := []backend.DataQuery{
+			{
+				JSON: json.RawMessage(`{
+				   "statistic":"Average",
+				   "period":"2h45m"
```

Comment:
Lol I had absolutely no idea we support this format in the editor. Thanks for adding a test for this1